src/components/ItemsList.js

```diff
@@ -1,38 +1,72 @@
 import React from 'react';
 import firebase from '../lib/firebase';
 import Nav from './Nav';
+import '../styles/ItemsList.css';
 import { useCollectionData } from 'react-firebase-hooks/firestore';
 
 const db = firebase.firestore().collection('shopping_list');
 
+const wasItemPurchasedWithinLastOneDay = (lastPurchasedOn) => {
```

Comment:
It works as intended on my end. I would love to know how  how you did the testing in firestore.

Reply:
That's great. Yes, it can be the fact that I'm not testing it correctly. Basically, I replaced `lastPurchasedOn` value in firestore console with another date (e.g Jan 31st) in milliseconds. (Jan 31st in milliseconds is `1612105921023 `).

Reply:
I would recommend you use previous date to simulate what is happening. Mark an item as purchased and replace the value of `lastPurchasedOn` by a date in milliseconds at least one day before today since the current date in milliseconds must always be greater than the value of `lastPurchasedOn`. E.g.
```js
const simulateLastPurchasedOn = Date.now() - (24 * 60 * 60 * 1000 + 1); // One day plus 1 milliseconds taken away from current date in milliseconds.
```
And then edit the value of one of the items you have marked as purchased, replacing it with `simulateLastPurchasedOn`.

Reply:
Ah, that's right. I was using the next day as the `lastPurchasedOn`. I just tested as you suggested and it works perfectly. Thank you Joseph!

Reply:
> I just tested if the item is disabled for 24 hours after the purchase, using firestore console. However, the item still stays disabled, after 24 hours are passed. Can you please check if the functionality is there? This could totally be an issue on my end. Thank you :)

I have just tested what you are talking about. It still gives me the intended outcome (I really don't know why yet) though it is like saying today is Jan 30, 2020 and the user has  marked the item as purchased on Jan 31, 2020 ( in the future ).

Reply:
@nibble0101 Love the investigation here, Joseph! Great work, and awesome demonstration of digging deeper to understand why something is behaving as it is 🙌

Reply:
@nibble0101 Yes, exactly. It works as expected now. I tested it as a future date which returned true in the boolean logic. Thanks for clarifying it. 👍